File: aicall-backend/src/main/java/com/aicall/module/evaluation/entity/Evaluation.java
package com.aicall.module.evaluation.entity;

import lombok.Data;
import java.time.LocalDateTime;

@Data
public class Evaluation {
    private Long id;
    private Long consultationId;
    private Long patientId;
    private Integer doctorScore;
    private Integer serviceScore;
    private String comment;
    private LocalDateTime createTime;
    private String consultationNo;
}

File: frontend/packages/shared/src/api/evaluation.ts
import { get, put } from './request';

export interface EvaluationItem {
  id: number;
  consultationId: number;
  doctorScore: number;
  serviceScore: number;
  comment: string;
  createTime: string;
  consultationNo: string;
}

export function getPendingEvaluations() {
  return get<EvaluationItem[]>('/user/evaluation/pending');
}

export function submitEvaluation(consultationId: number, data: { doctorScore: number; serviceScore: number; comment: string }) {
  return put<void>(`/user/evaluation/${consultationId}`, data);
}
